pkg/kms/crypto_box.go

```diff
@@ -133,7 +151,9 @@ func (b *CryptoBox) SealOpen(cipherText, myPub []byte) ([]byte, error) {
 		return nil, errors.New("message too short")
 	}
 
-	kp, err := b.km.getKeyPairSet(base58.Encode(myPub))
+	keySetID := hashKeySetID(base58.Encode(myPub))
```

Comment:
the detail of encoding format should be hidden from the rest of the code. 

we should not see so many calls to base58.encode.
rather we should see some variety of a KeyID function that hides this detail.

Reply:
that function question is hashKeyID() and hashKeySetID() ... it's not exposed as it should be only called within the kms package..

the encoding is an issue as sometimes we use key.Value which is already base58 encoded... if I move base58.Encode inside the ID function then everywhere we get the key from key.Value we will need to do base58.Decode()...  it may make more sense since Value is encoded but will end up doing Decode(Value) and then inside hashKeyID/hashKeySetID() we will do Encode(key) .. so we're duplicating instruction unnecessarily... this is why the encoding is kept for the caller to make.. again, this code is internal to kms and should not be exposed to the rest of the code.

Reply:
Even when this is internal functionality, we should still hide this detail.

Reply:
ok... will assume the base key argument is in raw bytes and do encoding inside the functions

Reply:
after refactoring, I noticed that the Packer interface expects base58 encoded keys, this clashes with hashKeyID refactoring of base58 keys to raw bytes keys, we can do an update of the interface and hashKey() in a later pr to expect raw bytes instead of base58 encoded key across the board.

For now, the expectation of verification keys stored in DID documents is to be base58 encoded, so public keys are expected to be encoded between calls in the framework. So it's safe to assume keys passed in arguments to be pre encoded.

Reply:
updated hashKeyID to take raw bytes keys.. I'll update hashKeySetID as well..

Reply:
> I noticed that the Packer interface expects base58 encoded keys, this clashes with hashKeyID refactoring of base58 keys to raw bytes keys, we can do an update of the interface and hashKey() in a later pr to expect raw bytes instead of base58 encoded key across the board.

The Packer interface expects raw byte keys, the Packager interface currently converts its base58 input to raw bytes for the Packer it invokes. It would simplify things to change the Packager's methods to take raw byte keys, and skip the conversion step.

Reply:
yes I'm aware of that... the rest of the code interacts with Packager while Packer is shielded by Packager (now the names are more confusing as both mean the same thing)

we can change this in a separate PR as this one is already too big

Reply:
done.. hashKeySetID now accepts raw key bytes instead of the base58 encoding of the key.

There are still a few places where the KMS API either returns base58 encoded keys like CreateKeySet() or accept base58 encoded key argument like SignMessage(). 

I have not changed these API functions as they have a wider impact on code outside of KMS. This can be done in a separate PR (if we choose to do it).